src/coreclr/src/vm/nativeimage.cpp

```diff
@@ -0,0 +1,156 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// --------------------------------------------------------------------------------
+// NativeImage.cpp
+//
+
+// --------------------------------------------------------------------------------
+
+#include "common.h"
+#include "nativeimage.h"
+
+// --------------------------------------------------------------------------------
+// Headers
+// --------------------------------------------------------------------------------
+
+#include <shlwapi.h>
+
+BOOL AssemblyNameIndexHashTraits::Equals(LPCUTF8 a, LPCUTF8 b)
+{
+    WRAPPER_NO_CONTRACT;
+
+    SString s1;
```

Comment:
`return SString(SString::Utf8, a).CompareCaseInsensitive(SString(SString::Utf8, b));`

Reply:
Actually, it can even be:
`return SString(SString::Utf8Literal, a).CompareCaseInsensitive(SString(SString::Utf8Literal, b));`

(saves unnecessary copy)

Reply:
Thanks for pointing that out, fixed in 16th commit.